action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -312,7 +322,11 @@ func (stateDB *StateDBAdapter) GetNonce(evmAddr common.Address) uint64 {
 		log.L().Error("Failed to get nonce.", zap.Error(err))
 		// stateDB.logError(err)
 	} else {
-		pendingNonce = state.PendingNonce()
+		if stateDB.useZeroNonceForFreshAccount {
+			pendingNonce = state.PendingNonceConsideringFreshAccount()
+		} else {
+			pendingNonce = state.PendingNonce()
+		}
```

Comment:
1. consider the case, where the legacy address send a first tx with `tx.Nonce=2`
here it returns `pendingNonce = 0`
2. inside EVM code, it calls `SetNonce(GetNonce()+1)` so it calls `SetNonce(1)`
3. then in our `SetNonce()`, it will hit 
```
if !stateDB.useConfirmedNonce {
		nonce--
}
```
`nonce=0`, and finally calls `SetPendingNonce(nonce+1)` --> `SetPendingNonce(1)`
4. `SetPendingNonce(1)` will trigger the conversion, convert the address to zero-nonce type with `s.nonce = 1`
5. but the first tx of this address now has `nonce=2`

Reply:
For Step 1, the tx.nonce is 0 if `state.PendingNonceConsideringFreshAccount()` is triggered in the EVM. I might ignore the case it's triggered when tx.nonce is 2. Could you give more details about how to reproduce?
For Step 5, What do you mean the address has `nonce=2`?

Reply:
the above steps illustrate a possible attack, someone just sends a tx with `tx.Nonce=2`, or a malicious delegate just run such a tx, when `state.PendingNonceConsideringFreshAccount()` is triggered in the EVM, it is `GetNonce() = 0`, it will pass `SetNonce(1)` and trigger the conversion
in other words, the first tx of such address will always trigger the conversion, regardless of the tx itself has `Nonce=2` or even `Nonce=5`
and this will be caught by `checkNonceContinuity`, but it becomes a potential attack